### DISCUSSION TITLE
Docker run always shows welcome page

### DISCUSSION THREAD

**Discussion by edank** [nuxt/nuxt#16540] [Questions] [answered]
Hi, my project loads my pages correctly when using npm run dev locally, or on a AWS elasticbeanstalk node instance.

When running from a docker image, all I see is the welcome page (regardless of the path).

Any ideas?

This is my current dockerfile

```
FROM node:16

WORKDIR /app

COPY package.json .

RUN npm install
RUN npm run build

COPY . .

EXPOSE 3000
EXPOSE 24678 
ENV HOST=0.0.0.0

CMD [ "node",".output/server/index.mjs"]

```

**Comment by edank** [comment, marked as answer]
Looks like it was lack of docker knowledge. I moved the 'Copy ..' command above the npm commands.

**Reply by luke-z** [reply]
You could put the `COPY . .` command in-between the two npm commands. 
For the first one, only the package.json file is required. But for the build you need all files.

Moving it between the two commands will lower build times, as packages won't be installed again if the package.json is the same :)

**Reply by edank** [reply]
Thanks. If I run 'copy . .' (which includes the package files) before 'npm install' why would that be shorter than if I only copied the package files? (assuming my node_modules folder is ignored of course)

**Reply by luke-z** [reply]
You will have to add a `.dockerignore` file to your folder structure, just like you would with a `.gitignore` file.

For example:
``` 
node_modules
.cache
Dockerfile
.dockerignore
.git
.env
build
```

Then you won't copy these files & folders to the image.

So the process will be:
1) add package.json
2) install packages
3) copy all other files (except those mentioned in the .dockerignore file)
4) build nuxt app

The Dockerfile would then look like that:
```
FROM node:16

WORKDIR /app

COPY package.json .
RUN npm install

COPY . .
RUN npm run build

EXPOSE 3000
EXPOSE 24678 
ENV HOST=0.0.0.0

CMD [ "node",".output/server/index.mjs"]
```

The benefit of doing it this way will be visible in build times of your images.
If you build the image from cache and you haven't changed anything in the package.json file, you won't need to reinstall all modules as they can be reused from cache.
